### DISCUSSION TITLE
individual plot via ctKalman

### DISCUSSION THREAD

**Discussion by SvenjaSchwarck** [cdriveraus/ctsem#58] [General]
Hi Charlie,

I wanted to plot 'y' and 'etasmooth' for every subject (so single plot per subject) using ctKalman.
My model was specified with two fully connected state variables enabling bi-directional coupling between PP and COG over 72 measurement occasions using the defaults (excepüt indvar for drift matrix):
model.output<-ctStanFit(datalong = session_data ctstanmodel = full_2CR.model ,  iter = 8000, chains = 4, plot = F, nopriors = F)

After plotting my data for all subjects using ctKalman I wanted to plot the data for every subject separately:
ctKalman(session.output, plot = T,subjects = 1, 
kalmanvec =c('y', 'etasmooth'), legend =T,   elementNames = c("RT_corrected", "Power_HR", "COG", "PP"))

For subjects = 1 it worked, but not for the other  16 subjects  (2:17), e.g. subjects = 2

I did that some month ago (using exactly the same code) and it worked for all subjects, now I only see a plot for subjects = 1, all the other plots are just white. 

My R version is 4.3.1, ctsem_3.9.0

Is there a change in ctKalman due to the newer ctsem version  that I'm overlooking?

Best and thanks,
Svenja

**Comment by cdriveraus** [comment]
Hi Svenja, there have been a couple of little bugs in the link between id's in your data, and the integers that ctsem uses internally, that occurred due to some recent changes in the Kalman related functions. ctsem 3.9.1 is the latest on CRAN, this might already fix your issue, or if not I just posted a github update in the last days that fixes an issue with factor id's in this context. If still no success let me know...

**Comment by SvenjaSchwarck** [comment]
It worked, thanks :)
